### PR TITLE
Mirror standard output plot class

### PR DESCRIPTION
**Pull request by braden6521**
- Added a StandardPlotOuptut class to the CSP directory. This produces standard formatted plots. 
- Added example and unit tests for StandardPlotOutput.
- Added processing steps to curvature plots when viewing orthorectified images of a mirror. This helps make otherwise unviewablely noisy images understandable.

### CODE REVIEW

File: opencsp/common/lib/csp/StandardPlotOutput.py [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+                    output_dir=self.options_file_output.output_dir,
+                    dpi=self.options_file_output.save_dpi,
+                    format='png',
+                    close_after_save=self.options_file_output.close_after_save,
+                )
+        else:
+            lt.info('Do not have both measured and reference optic; skipping slope deviation plots.')
+
+    def _plot_ray_trace_image_measured_optic(self):
+        """Plot ray trace image for measured optic"""
```
Comment (e10harvey) [inline opencsp/common/lib/csp/StandardPlotOutput.py:275]: no need for doc-strings here. if you don't want this to potentially be pulled into the online documentation, I would convert to hash-tag comment.

File: opencsp/common/lib/csp/StandardPlotOutput.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+            # Create ensquared energy curve
+            ee, ws = rt.ensquared_energy(ray_pts, self.options_ray_trace_vis.ensquared_energy_max_semi_width)
+
+            # Save data
+            self._ray_trace_output_reference = _RayTraceOutput(ray_trace, image, xv, yv, ee, ws)
+        else:
+            lt.info('No reference optic; skipping reference optic ray trace.')
+
+    def _ray_trace_scene(self, obj: RayTraceable) -> rt.RayTrace:
+        """Performs a raytrace of a simple scene with a source and an optic.
```
Comment (e10harvey) [inline opencsp/common/lib/csp/StandardPlotOutput.py:527]: Same as above. I'm fine with leaving these as doc-strings, but don't want to give the impression that this is a user-facing function.

**Review by e10harvey — APPROVED** [review]
In general everything looks great. I would again advocate for random values to be used in the tests, rather than hard coded numbers like 0, 7, and 100. The only issue I see is with using doc-strings rather than hash-tag comments for protected class members. If this is no longer a standard for us, we should clarify that in https://github.com/sandialabs/OpenCSP/blob/develop/doc/source/contributing.rst#coding-standards.

**Comment by braden6521** [comment]
@e10harvey. Thanks for your review. I've made the following updates, could you please re-review?

- I updated the docstrings to be hashtags for private members. 
- On that note, I also added more documentation to the StandardPlotOutput class itself (there were a few places where I thought it was lacking).
- I also added some more checks to the unit test. 
- I'm not sure how to add randomness into the unit test. Since we are comparing generated images to saved images, the inputs have to be the same every time. I can see cases where this could be employed in the future though on other tests.

File: opencsp/common/lib/csp/StandardPlotOutput.py [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+        if (not self._has_reference_ray_trace) and (not self._has_measured_ray_trace):
+            lt.info('No measured or reference ray trace data; skipping enclosed energy plot.')
+            return
+
+        # Make figure
+        fig_rec = fm.setup_figure(
+            self.fig_control, name='Ensquared Energy', number_in_name=self.options_file_output.number_in_name
+        )
+
+        # Dray reference if available
```
Comment (e10harvey) [inline opencsp/common/lib/csp/StandardPlotOutput.py:352]:
```suggestion
        # Draw reference if available
```

**Review by e10harvey — DISMISSED** [review]
Thank you, @braden6521 !

**Comment by braden6521** [comment]
Thanks @e10harvey!

**Pull request merged**